pipeline-runner/R/qc-6-integrate_scdata-seuratv4.R

```diff
@@ -31,18 +31,13 @@ run_seuratv4 <- function(scdata_list, config, cells_id) {
   # use the min of what the user wants and what can be calculated
   npcs <- min(config$dimensionalityReduction$numPCs, npcs_for_pca)
 
-  scdata_list <- order_by_size(scdata_list)
+  scdata_list <- prepare_scdata_list_for_seurat_integration(scdata_list, config, cells_id)
 
   # normalize single samples
   for (i in 1:length(scdata_list)) {
     # we need RNA assay to compute the integrated matrix
     Seurat::DefaultAssay(scdata_list[[i]]) <- "RNA"
 
-    # remove cell cycle genes if needed
-    if (length(exclude_groups) > 0) {
-      scdata_list[[i]] <- remove_genes(scdata_list[[i]], exclude_groups)
-    }
-
     if (normalization == "LogNormalize") {
```

Comment:
Shouldn't the defaultassay <- RNA be a part of the prepare list function?

Do we even need the defaultAssay <- RNA?

Reply:
not sure if we need it. I followed the same structure as harmony integration.

Reply:
My concern is that remove_genes subsets:

 scdata <- subset(scdata, features = keep_genes)

If we need to fix the defaultAssay to RNA, that means that this subset might *not* be applied on the RNA assay at this stage.